types/web3/index.d.ts

```diff
@@ -29,7 +29,7 @@ import { BatchRequest, Net, Personal } from "./eth/types";
 import Utils from "./utils";
 import Eth from "./eth/index";
 
-declare class Web3 {
+export declare class Web3 {
```

Comment:
this is only compatible with web3.js 2.x because esm is available:

https://github.com/ethereum/web3.js/blob/2.x/packages/web3/package.json#L12